### ISSUE TITLE
Allow class expressions with max-classes-per-file

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#3270]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```typescript
export abstract class UnsupportedVisitor extends NodeVisitor {
    public static withDescriptor(descriptor: string): typeof UnsupportedVisitor {
        return class extends UnsupportedVisitor {
            protected descriptor = descriptor;
        };
    }

    protected abstract descriptor: string;

    public visit(node: Node): undefined {
        return `${this.descriptor} is not supported.`;
    }
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "max-classes-per-file": true
    }
}
```

#### Actual behavior

A TSLint complaint is logged for the `class extends` node.

#### Expected behavior

I would hope for a way to exclude class _expressions_ (rather than declarations) from the rule.

**Comment by ajafff** [comment]
Fixed by #3281
